### ISSUE TITLE
[Code structure] Broken highlighting after certain edits

### ISSUE THREAD

**Issue by microbit-matt-hillsdon** [microbit-foundation/python-editor-next#259]
Delete and reinstate the colon after "happy" in the (partially minimised) code below. Observe highlighting is broken.
It is sensitive to comment length (e.g. you can delete some but not all of the "random 99" comment). A CM incremental editing bug?

```
from microbit import *
import music
from random import *
program_list = []
global program_counter
program_counter = -1
variables = [None]*26
operators = '+-/*'

def parse(instruction):
    global program_counter
#    print(program_counter)
    if instruction.startswith('goto '):
        a = 1
    elif instruction == 'stop':
        a = 1
    elif instruction.startswith('beep '):
        a = 1
    elif instruction.startswith('random '):
        # random a 99 puts a random number between 1 and 99 in variable a
        var_rnd = instruction[7]
        var_index = ord(var_rnd) - 97
        rnd_range = int(instruction[9:])
        variables[var_index] = randint(1,rnd_range)
    elif instruction.startswith('if '):
        var1 = ord(instruction[3]) - 97
        operator = instruction[4]
        var2 = ord(instruction[5]) - 97
        goto = int(instruction[12:])
        if goto  < 0 or goto > len(program_list):
            print('Goto error: line doesn\'t exist.')
        else:
            if operator == '=':
                if variables[var1] == variables[var2]:
                    program_counter = goto - 1
            elif operator == '>':
                if variables[var1] > variables[var2]:
                    program_counter = goto - 1
            elif operator == '<':
                if variables[var1] < variables[var2]:
                    program_counter = goto - 1
            else:
                print('If error: comparsion operators must be =, < or >')
    elif instruction.startswith('scroll '):
        split = instruction.find(' ') + 1
        display.scroll(instruction[split:])
    elif instruction.startswith('input ') and len(instruction) == 7:
        var_input = instruction[6]
        var_index = ord(var_input) - 97
        if var_index >= 0 and var_index <= 26:
            input_text = input()
            try:
                variables[var_index] = int(input_text)
            except ValueError:
                variables[var_index] = input_text
        else:
            print('Variable names must be letter a-z.')
    elif instruction.startswith('print '):
        if len(instruction) == 7 and instruction[6].isalpha():
            var_print = instruction[6]
            var_index = ord(var_print) - 97            
            print(variables[var_index])
        else:
            split = instruction.find(' ') + 1
            print(instruction[split:])
    elif instruction[3] in operators and instruction[1] == '=' and ord(instruction[0]) > 96 and ord(instruction[0]) < 123 and ord(instruction[2]) > 96 and ord(instruction[2]) < 123 and ord(instruction[4]) > 96 and ord(instruction[4]) < 123:
        var1 = instruction[0]
        var2 = instruction[2]
        var3 = instruction[4]
        operator = instruction[3]
        if operator == '+':
            variables[ord(var1)-97] = 1
        elif operator == '-':
            variables[ord(var1)-97] = 1
        if operator == '/':
            variables[ord(var1)-97] = 1
        if operator == '*':
            variables[ord(var1)-97] = 1
    elif '=' in instruction:
        split = instruction.find('=') + 1
        var_name = instruction[:split-1]
        var_contents = instruction[split:]
        if len(var_name) == 1 and var_name.isalpha():
            try:
                variables[ord(var_name)-97] = 1#nt(var_contents)
            except ValueError:
                variables[ord(var_name)-97] = var_contents
#            print(var_name, '=', var_contents)
        else:
            print('Variable names must be one letter a-z.')
    elif instruction == 'heart':
        display.show(Image.HEART)
    elif instruction == 'happy':
        display.show(Image.HAPPY)
    elif instruction == 'sad':
        display.show(Image.SAD)
    elif instruction == 'meh':
        display.show(Image.MEH)
    elif instruction == 'confused':
        display.show(Image.CONFUSED)
    elif instruction == 'angry':
        display.show(Image.ANGRY)
    elif instruction == 'asleep':
        display.show(Image.ASLEEP)
    elif instruction == 'yes':
        display.show(Image.YES)
    else:
        display.show('?')
```

**Comment by microbit-matt-hillsdon** [comment]
The outstanding CM bug has been fixed. Matt to review and plan upgrade. For independent reasons there's already an upgrade PR but that hit unrelated issues.

**Comment by microbit-matt-hillsdon** [comment]
Hopefully fixed by #563

**Comment by microbit-robert** [comment]
Fix in #563. Closing.
